File: bet-processing/src/main/scala/com/verity/arbitrage/model/MarketOdds.scala
package com.verity.arbitrage.model

import com.verity.arbitrage.ArbitrageCalculator.{getBestOddsEachTeam, getMarketMargin}

case class MarketOdds(
                     id: String,
                       sportKey: String,
                       sportNice: String,
                       teams: Seq[String],
                       commenceTime: Long,
                       homeTeam: String,
                       odds: Seq[CompetitorOdds],
                       bestMargin: Double,
                      bestOdds: Seq[CompetitorOdds]
                     ) {
  def getBestMargin: Double = getMarketMargin(getBestOddsEachTeam(odds))
}

File: webapp/ui/src/app/models/market-odds.ts
import { CompetitorOdds } from "./competitor-odds";

export interface MarketOdds{
  id: string;
  sportKey: string;
  sportNice: string;
  teams: Array<string>;
  commenceTime: number;
  homeTeam: string;
  odds: Array<CompetitorOdds>;
  bestMargin: number;
}
